### DISCUSSION TITLE
updateMany many-to-many relations

### DISCUSSION THREAD

**Discussion by ilya-lopukhin** [prisma/prisma#2231] [Q&A] [answered]
Hey guys, i have a situation here where i feel like i hit some prisma limitation i can't workaround yet. I got a simple chat and i need to mark messages as "read" by particular users to mark it visually for other users. When i first approached this, i thought it's as simple as updating the `seenBy` relations on a message model

```
  updateManyMessages(where: { chat: { id: $chatId }, data: { seenBy: { connect: { id: $userId } } }) { 
    count
  }
```
But i quickly realized it's not possible with prisma, and updating every single message in a loop just hangs the prisma for me if there are more than 100 messages. Any advice on how should i implement functionality like this?

Also, wanted to understand why prisma adds non-scalar fields to `UpdateMany` inputs if they don't take any effect at all? Can we consider it as an error during inputs generation if it should not be possible to update many many-to-many relations?

**Comment by ryands17** [comment, marked as answer]
Hey @ilya-lopukhin :wave: 
Did you try writing it like this?

```ts
prisma.chat.update({
    where: { id: 1 },
    data: {
      seenBy: {
        connect: [{ id: 1 }, { id: 2 }, { id: 3 }],
      },
    },
})
```

This will update the seen for all users specified in connect

**Reply by ilya-lopukhin** [reply]
While you're answer is kinda irrelevant to my question it lead me to a powerful enlightenment that i should just connect many messages to a user, meaning i need to
```
updateUser(where: { id: "..." }, data: { messagesSeen: { connect: [{ id: "" }, { id: "" }] } }) { id }
```

and thats it. Simple things sometimes not so obvious, thanks a lot for bringing me to this point

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
